### DISCUSSION TITLE
Are the MLRs sent automatically?

### DISCUSSION THREAD

**Discussion by vinoddanekar** [OxalisCommunity/oxalis#591] [General]
Hi all,

I have a question: Using Oxalis AS4, are the MLRs (Message Level Responses) sent automatically to the sender or one needs to crawl, validate and send MLR to senders AP?

Thanks in advance.

Vinod

**Comment by aaron-kumar** [comment]
@vinoddanekar : Receiving AP themselves need to validate incoming document, generate [MLR](https://docs.peppol.eu/poacc/upgrade-3/profiles/36-mlr/) (ApplicationResponse) and then send it back to sending AP provided receiver/participant registered with MLR document type in SMP.  

Oxalis only handle Transport Level Response/validation and it do **N**ot yet support Message Level Response/validation or MLR generation at this moment. 
If you are regular/paid Oxalis member then please raise "feature request".  If not, please look forward to join Oxalis as Gold or Regular member.

**Reply by vinoddanekar** [reply]
Hi Aaron,

Thank you so much for information.

Vinod.
